app/models/piece.rb

```diff
@@ -16,8 +16,15 @@ def is_obstructed?(x,y)
 	end
 
 	def move_to(x,y)
-		update_attributes(x_position: x, y_position: y)
+      updated_status = update_attributes(x_position: x, y_position: y)
+      self.moved = true if updated_status
 	end
+  
+    def valid_move?(x, y)
+      # Valid parameters passed in?
+      return false if x < 0 || y < 0 || x > 7 || y > 7
```

Comment:
this is me being a bit being a pain in the **- and continuing our conversation with Walley on slack:

I would suggest to put this logic in a private method like outside_board?(x,y) which has all this if statements.. this way you immediately see what it does in plain english- just so much easier to read and understand the code for someone who hasn't been thinking about chess in the last weeks :)